src/main/drivers/pwm_output.c

```diff
@@ -403,11 +432,61 @@ void pwmWriteDshotCommand(uint8_t index, uint8_t motorCount, uint8_t command)
             }
 
             pwmCompleteDshotMotorUpdate(0);
-            delay(1);
+        }
+        delayMicroseconds(timeDelayUs);
+    } else {
+        for (uint8_t i = 0; i < motorCount; i++) {
+            if ((i == index) || (index == ALL_MOTORS)) {
+                if (dshotCommandControl.command[i] == 0) {
+                    dshotCommandControl.command[i] = command;
+                    dshotCommandControl.repeats = repeats;
+                    dshotCommandControl.nextCommandAt = timeNowUs + DSHOT_COMMAND_DELAY_US;
+                    dshotCommandControl.delayAfterCommand = timeDelayUs;
+                }
+            }
         }
     }
 }
 
+FAST_RAM bool pwmIsProcessingDshotCommand(void)
+{
+    return dshotCommandControl.nextCommandAt;
+}
+
+uint8_t pwmGetDshotCommand(uint8_t index)
+{
+    return dshotCommandControl.command[index];
+}
+
+bool pwmProcessDshotCommand(uint8_t motorCount)
```

Comment:
This needs to be `bool NOINLINE pwmProcessDshotCommand(uint8_t motorCount)`, to make sure it does not get inlined into the scarce ITCM_RAM by the compiler.

Reply:
@mikeller: that will also disable optimization on targets without ITCM_RAM .. Better name than `NOINLINE` should be used ...